doc/sphinx/user-extensions/syntax-extensions.rst

```diff
@@ -215,15 +215,22 @@ definition is the following:
 More generally, it is required that notations are explicitly factorized on the
 left. See the next section for more about factorization.
 
+.. _NotationFactorization:
+
 Simple factorization rules
 ~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 Coq extensible parsing is performed by *Camlp5* which is essentially a LL1
 parser: it decides which notation to parse by looking at tokens from left to right.
 Hence, some care has to be taken not to hide already existing rules by new
-rules. Some simple left factorization work has to be done. Here is an example.
+rules. Indeed, notations with a common prefix setting different levels could
+interfere with one another. For instance, a notation ``x < y`` with ``x``
+and ``y`` at level 69 would be broken by a further rule that would put
+``y`` at another level, like ``x < y < z`` with ``x`` at level 69 and ``y``
+at level 200. To avoid such issues, one should left factorize rules, that is ensure
+that common prefixes set the same levels.
```

Comment:
```suggestion
rules. Indeed, notations with a common prefix but different levels can
interfere with one another, causing some of the them to parse incorrectly.  For instance, a notation ``x < y`` with ``x``
and ``y`` at level 69 would be broken by another rule that puts
``y`` at another level, like ``x < y < z`` with ``x`` at level 69 and ``y``
at level 200. To avoid such issues, you should left factorize rules, that is ensure
that common prefixes use the same levels.
```

Should it be "can interfere" or "will interfere"?

Reply:
Can interfere, since in some articular cases both notation can work fine.
What about "notations with a common prefix but different levels can
interfere with one another, causing some of them to not parse."? (since what actually happens is that one of the notations (sometimes both) becomes unusable)

Reply:
How about "not to work"?  or (better) "makes some of them unusable"?
(probably gives parse errors but the doc doesn't need to be that specific).